windows/CodePushPackage.cs

```diff
@@ -0,0 +1,364 @@
+                }
+                else
+                {
+                    if (diffManifestFile != null)
+                    {
+                        // TODO verify hash for diff update
+                        // CodePushUpdateUtils.verifyHashForDiffUpdate(newUpdateFolderPath, newUpdateHash);
+                    }
+
+                    updatePackage[RELATIVE_BUNDLE_PATH_KEY] = relativeBundlePath;
+                }
+            }
+            catch (InvalidDataException)
+            {
+                // Downloaded file is not a zip, assume it is a jsbundle
+                await downloadFile.RenameAsync(expectedBundleFileName);
+                await downloadFile.MoveAsync(newUpdateFolder);
+            }
+
+            // Save metadata to the folder
+            await FileIO.WriteTextAsync(newUpdateMetadataFile, JsonConvert.SerializeObject(updatePackage));
+        }
+
+        public async Task InstallPackage(JObject updatePackage, bool removePendingUpdate)
+        {
+            string packageHash = (string)updatePackage[PACKAGE_HASH_KEY];
+            JObject info = await GetCurrentPackageInfo();
+            if (removePendingUpdate)
+            {
+                StorageFolder currentPackageFolder = await GetCurrentPackageFolder();
+                if (currentPackageFolder != null)
+                {
+                    await currentPackageFolder.DeleteAsync();
+                }
+            }
+            else
+            {
+                string previousPackageHash = await GetPreviousPackageHash();
+                if (previousPackageHash != null && !previousPackageHash.Equals(packageHash))
+                {
+                    StorageFolder previousPackageFolder = await GetPackageFolder(previousPackageHash, false);
+                    if (previousPackageFolder != null)
+                    {
+                        await previousPackageFolder.DeleteAsync();
+                    }
+                }
+
+                info[PREVIOUS_PACKAGE_KEY] = info[CURRENT_PACKAGE_KEY];
+            }
+
+            info[CURRENT_PACKAGE_KEY] = packageHash;
+            await UpdateCurrentPackageInfo(info);
+        }
+
+        public async Task RollbackPackage()
+        {
+            JObject info = await GetCurrentPackageInfo();
+            StorageFolder currentPackageFolder = await GetCurrentPackageFolder();
+            if (currentPackageFolder != null)
+            {
+                await currentPackageFolder.DeleteAsync();
+            }
+
+            info[CURRENT_PACKAGE_KEY] = info[PREVIOUS_PACKAGE_KEY];
+            info[PREVIOUS_PACKAGE_KEY] = null;
+            await UpdateCurrentPackageInfo(info);
+        }
+
+        public void DownloadAndReplaceCurrentBundle(string remoteBundleUrl, string bundleFileName)
+        {
+            // TODO implement this method (only used in tests)
```

Comment:
if this is only used in your ios tests that you made for the react native ios tests, then i'm probably going to remove it so u shouldn't worry about having to implement this